openprovider/openprovider.php

```diff
@@ -1566,24 +1603,48 @@ private function manageSettings(
                 $this->logRequest($api);
 
                 if ($update_is_locked_domain_response->getCode() != 0) {
-                    $this->Input->setErrors([
-                        'errors' => [
-                            $update_is_locked_domain_response->getMessage() . ': ' . $update_is_locked_domain_response->getCode()
-                        ]
-                    ]);
+                    $this->showError($update_is_locked_domain_response->getMessage() . ': ' . $update_is_locked_domain_response->getCode());
                 } else {
-                    $op_domain['is_locked'] = $domain_transfer_lock;
+                    $vars->is_locked = isset($domain_transfer_lock) && $domain_transfer_lock ? 'true' : 'false';
                 }
             }
         }
 
-        if (isset($op_domain['auth_code']) && $op_domain['auth_code']) {
-            $vars->epp = $op_domain['auth_code'];
-        }
+       return $this->showView($vars);
+    }
 
-        $vars->is_locked = isset($op_domain['is_locked']) && $op_domain['is_locked'] ? 'true' : 'false';
+    /**
+     * $this->Input should be initialized for this method to be executed correctly
+     *
+     * @param string $error_message
+     *
+     * @return bool false
+     */
+    private function showError(string $error_message): bool
+    {
+        $this->Input->setErrors([
+            'errors' => [
+                $error_message,
+            ]
+        ]);
 
-        $this->view->set('vars', $vars);
+        return false;
+    }
+
+    /**
+     * $this->view should be initialized and template should be configured before you can execute this method
+     *
+     * @param object|null $vars
+     *
+     * @return string HTML string from template
+     *
+     * @throws Exception
+     */
+    private function showView($vars = null): string
```

Comment:
// Render template and return results as a string.
render($vars = null): string;